### PR TITLE
fix: do not inject dispatch prop when using the enhancer

### PR DESCRIPTION
**Pull request by satazor**
The reason is that we might use the enhancer to translate stuff inside a component that also uses `...restProps` to passdown props to a DOM element.
Previously we had to declare the `dispatch` prop in the component just omit it:

```js
const SomeComponent = ({ p, dispatch, ...restProps }) => <div { ...restProps }>{ p.t('foo') }</div>;

SomeComponent.propTypes = {
   p: PropTypes.object.isRequired,
   dispatch: PropTypes.func.isRequired,
};

export default translate(SomeComponent);
```

With this change, it's no longer necessary to do that:

```js
const SomeComponent = ({ p, ...restProps }) => <div { ...restProps }>{ p.t('foo') }</div>;

SomeComponent.propTypes = {
   p: PropTypes.object.isRequired,
};

export default translate(SomeComponent);
```

Unrelated to this PR: `prop-types` dependency was missing

### CODE REVIEW

**Comment by satazor** [comment]
@guillaumearm Do you have some time to take a look at this?

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
   "dependencies": {
     "lodash.curry": "^4.1.1",
     "node-polyglot": "^2.2.2",
+    "prop-types": "^15.6.1",
```
Comment (guillaumearm) [inline package.json]: `prop-types` should be a peerDependency.
Anyway, this should be done in another PR, can you revert `package.json` and `package-lock.json` ?

**Review by guillaumearm — CHANGES_REQUESTED** [review]
Sorry for the late @satazor, just a thing to fix before merging this PR.

File: package.json (hunk 1 above)
Comment (satazor) [reply package.json]: Could you elaborate why it should be a peer dependency?

If it's a peer dependency, then it should also be declared in the devDependencies so that tests pass.

Comment (guillaumearm) [reply package.json]: > Could you elaborate why it should be a peer dependency?

This should be a peer dependency because it intended to be used in the user project.
When I say `it`, I'm talking about `react`, `redux` and `react-redux`. 

When you use `redux-polyglot`, you have to install `react`, `redux` and `react-redux` yourself.

It's the same for `react-redux` (please see [react-redux peerDependencies](https://github.com/reactjs/react-redux/blob/master/package.json#L46)).

Refs : 
  - https://nodejs.org/en/blog/npm/peer-dependencies/
  - https://stackoverflow.com/questions/26737819/why-use-peer-dependencies-in-npm-for-plugins?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa


-----------------

> If it's a peer dependency, then it should also be declared in the devDependencies so that tests pass.

Yes, you right, so you can add `prop-types` in devdep and peerdep.

------------------

Anyway, Let's finish this PR and discuss about peer deps in #94.

Comment (satazor) [reply package.json]: Done, I've moved as a peer dependency.

Regarding the existing peer dependencies, should we make react `^15.0.0` instead of `^15.5.4` and the same for react-redux?

**Review by guillaumearm — APPROVED** [review]

**Comment by guillaumearm** [comment]
`redux-polyglot@0.6.4` is out 🎉
